lib/dialects/abstract/query-generator.js

```diff
@@ -1342,7 +1342,9 @@ const QueryGenerator = {
         ].join(' '));
 
         if (_.isPlainObject(topLevelInfo.options.where)) {
-          topLevelInfo.options.where['__' + includeAs] = subQueryWhere;
+          //includeAs can be a string or an object
+          const key = includeAs && includeAs.externalAs || includeAs;
```

Comment:
if it's a non-empty string, this will evaluate to `includeAs.externalAs`, which is `undefined`